service/src/test/java/bio/terra/workspace/service/workspace/ApplicationServiceTest.java

```diff
@@ -0,0 +1,201 @@
+// Create workspace
+// Enumerate/get
+// enable operating - works
+// enable deprecated - fails
+// enableNocheck deprecated - works
+// Enumerate/get
+// disable deprecated
+// disable operating
+// Enumerate/get
+
+public class ApplicationServiceTest extends BaseUnitTest {
+  private static final UUID LEO_UUID = UUID.fromString("4BD1D59D-5827-4375-A41D-BBC65919F269");
+  private static final UUID CARMEN_UUID = UUID.fromString("EB9D37F5-BAD7-4951-AE9A-86B3F03F4DD7");
+  private static final UUID NORM_UUID = UUID.fromString("DB186A44-2881-4CD6-9D1B-65441574C3B1");
+  private static final UUID UNKNOWN_UUID = UUID.fromString("DB186A44-2881-4CD6-9D1B-65441574C3B2");
+
+
+  private static final WsmApplication LEO_APP =
+      new WsmApplication()
+          .applicationId(LEO_UUID)
+          .displayName("Leo")
+          .description("application execution framework")
+          .serviceAccount("leo@terra-dev.iam.gserviceaccount.com")
+          .state(WsmApplicationState.OPERATING);
+
+  private static final WsmApplication CARMEN_APP =
+      new WsmApplication()
+          .applicationId(CARMEN_UUID)
+          .displayName("Carmen")
+          .description("musical performance framework")
+          .serviceAccount("carmen@terra-dev.iam.gserviceaccount.com")
+          .state(WsmApplicationState.DEPRECATED);
+
+  private static final WsmApplication NORM_APP =
+      new WsmApplication()
+          .applicationId(NORM_UUID)
+          .displayName("Norm")
+          .description("old house framework")
+          .serviceAccount("norm@terra-dev.iam.gserviceaccount.com")
+          .state(WsmApplicationState.DECOMMISSIONED);
+
+  /** A fake authenticated user request. */
+  private static final AuthenticatedUserRequest USER_REQUEST =
+      new AuthenticatedUserRequest()
+          .token(Optional.of("fake-token"))
+          .email("fake@email.com")
+          .subjectId("fakeID123");
+
+  @Autowired ApplicationDao appDao;
+  @Autowired WsmApplicationService appService;
+  @Autowired WorkspaceService workspaceService;
+  @Autowired JobService jobService;
+
+  /** Mock SamService does nothing for all calls that would throw if unauthorized. */
+  @MockBean private SamService mockSamService;
+
+  private UUID workspaceId;
+
+  @BeforeEach
+  void setup() throws Exception {
+    // Set up so all spend profile and workspace checks are successful
+    Mockito.when(
+        mockSamService.isAuthorized(
+            Mockito.any(),
+            Mockito.eq(SamConstants.SPEND_PROFILE_RESOURCE),
+            Mockito.any(),
+            Mockito.eq(SamConstants.SPEND_PROFILE_LINK_ACTION)))
+        .thenReturn(true);
+    Mockito.when(
+        mockSamService.isAuthorized(
+            Mockito.any(),
+            Mockito.eq(SamConstants.SAM_WORKSPACE_RESOURCE),
+            Mockito.any(),
+            Mockito.any()))
+        .thenReturn(true);
+
+    // Return a valid google group for cloud sync, as Google validates groups added to GCP projects.
```

Comment:
This is a unit test, it shouldn't be talking to GCP. Is this in anticipation for creating application-owned resources?

Reply:
No, I copied that mock from WorkspaceServiceTest which is a connected test.
I probably don't need it. CreateWorkspace doesn't call sync, right?
I just want the create workspace to succeed so I can exercise the application service.